aiida/cmdline/params/options/callable.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""
+.. py:module::callable
+    :synopsis: A monkey-patched subclass of click.Option that does not evaluate callable default during tab completion
+"""
+
+import typing as t
+
+import click
+
+__all__ = ('CallableDefaultOption',)
+
+
+class CallableDefaultOption(click.Option):
+    """A monkeypatch for click.Option that does not evaluate default callbacks during tab completion
+
+    This is a temporary solution until a proper fix is implemented in click, see:
+    https://github.com/pallets/click/issues/2614
+    """
+
+    def get_default(self, ctx: click.Context, call: bool = True) -> t.Optional[t.Union[t.Any, t.Callable[[], t.Any]]]:
+        """provides the functionality of :meth:`click.Option.get_default`,
+        but ensures we do not evaluate callable defaults when in tab-completion context."""
+        if ctx.resilient_parsing:
+            # This is done in interactive.py:InteractiveOption as well
+            return None
+            # Another possibility
+            # return super().get_default(ctx=ctx, call=False)
```

Comment:
```suggestion
        if ctx.resilient_parsing:
            return None
```
I would just keep the version that we also keep in the InteractiveOption. Looked at the source and the `Option` class has some additional logic for non-boolean flags that does not seem to take `ctx.resilient_parsing` into account. Unlikely we would hit this, but I think returning `None` does the job and prevents any other complexity